op-service/dial/dial.go

```diff
@@ -21,8 +21,8 @@ const defaultRetryTime = 2 * time.Second
 // DialEthClientWithTimeout attempts to dial the L1 provider using the provided
 // URL. If the dial doesn't complete within defaultDialTimeout seconds, this
 // method will return an error.
-func DialEthClientWithTimeout(timeout time.Duration, log log.Logger, url string) (*ethclient.Client, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+func DialEthClientWithTimeout(ctx context.Context, timeout time.Duration, log log.Logger, url string) (*ethclient.Client, error) {
```

Comment:
Once you've added the context, there's not much point of retaining the timeout as callers can set the timeout

Reply:
Good point, I'll refactor it as part of the remaining op-proposer work, when the remaining callers of this method handle their contexts better here